config/v1/types_infrastructure.go

```diff
@@ -570,6 +581,40 @@ type EquinixMetalPlatformStatus struct {
 	IngressIP string `json:"ingressIP,omitempty"`
 }
 
+// AlibabaCloudPlatformSpec holds the desired state of the Alibaba Cloud infrastructure provider.
+// This only includes fields that can be modified in the cluster.
+type AlibabaCloudPlatformSpec struct{}
+
+// AlibabaCloudPlatformStatus holds the current status of the Alibaba Cloud infrastructure provider.
+type AlibabaCloudPlatformStatus struct {
+	// region specifies the region for Alibaba Cloud resources created for the cluster.
+	// +kubebuilder:validation:Pattern=`^[0-9A-Za-z-]+$`
+	Region string `json:"region"`
```

Comment:
noticing the explicit `+required` down on `AlibabaCloudResourceTag.Key` makes me wonder why we don't have a policy requiring an explicit `+optional` or `+required` on each property.  Is there some kubebuilder setting that makes the default mutable at the type level?  Not clear to me why we set `+required` on some properties, `+optional` on others, and leave still more implicit.

Reply:
yes, some other infrastructure providers above wrote their codes with some confused tag using. if it's ok i'll add `+required` tag for all fields can not be empty.

Reply:
you must use `// +kubebuilder:validation:Required`. `+required` is not supported by kubebuilder.